build.sbt

```diff
@@ -20,7 +20,7 @@ val ghTokenSource = TokenSource.GitConfig("github.token") || TokenSource.Environ
 githubTokenSource := ghTokenSource
 
 lazy val commonSettings = Seq(
-  scalacOptions ++= List("-Ymacro-annotations", "-Yrangepos", "-Wconf:cat=unused:info", "-language:reflectiveCalls"),
+  scalacOptions ++= List("-Ymacro-annotations", "-Yrangepos", "-Wconf:cat=unused:info", "-language:reflectiveCalls", "-Wconf:msg=While parsing annotations in:silent"),
```

Comment:
Why do we need this? Does removing `guava` enforce adding this `Wconf` parameter?

Reply:
It's needed to avoid this error:
```
[error] While parsing annotations in /Users/effects-ai/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/io/micrometer/micrometer-commons/1.10.2/micrometer-commons-1.10.2.jar(io/micrometer/common/lang/Nullable.class), could not find MAYBE in enum <none>.
[error] This is likely due to an implementation restriction: an annotation argument cannot refer to a member of the annotated class (scala/bug#7014).
[error] No warnings can be incurred under -Werror.
[error] two errors found
```

Reply:
Ok, have you seen this error before removing guava? If so, it should be a separate change, not together with removing guava. Otherwise, it would be impossible to revert if needed. Also, it would mean it's your specific setup improperly configured, as everything is working correctly for everybody else over a long time.

Reply:
btw:
```
[error] two errors found
```

What are those?

Reply:
> ```
> [error] While parsing annotations in /Users/effects-ai/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/io/micrometer/micrometer-commons/1.10.2/micrometer-commons-1.10.2.jar(io/micrometer/common/lang/Nullable.class), could not find MAYBE in enum <none>.
> [error] This is likely due to an implementation restriction: an annotation argument cannot refer to a member of the annotated class (scala/bug#7014).
> [error] No warnings can be incurred under -Werror.
> ```

Reply:
> Ok, have you seen this error before removing guava? If so, it should be a separate change, not together with removing guava. Otherwise, it would be impossible to revert if needed. Also, it would mean it's your specific setup improperly configured, as everything is working correctly for everybody else over a long time.

I'll try to setup a completely different setup to verify.

Reply:
It reproducible on a clean install of Ubuntu 22.0 as well.

This is another reason to drop guava - it hides issues with other dependencies. This issue affects all versions of `micrometer` up to 1.12, micrometer 1.13 seems to be fixed, but it requires code changes in `Metrics.scala`.